### DISCUSSION TITLE
SSH Timeout, Azure VM Linux

### DISCUSSION THREAD

**Discussion by milons** [AzureCommunityPL/group#22] [Pomoc] [answered]
Cześć. 
Nie mogę dostać się na VM postawioną na Azure przez SSH (Linux). 
Maszyna wykonała restart, nie do końca wiem dlaczego (log poniżej), po czym żadna z usług na niej działających nie jest dostępna, a przez SSH nie dam rady się zalogować (Timeout). 
Myślałem, że dysk jest pełny, ale po pobraniu widzę, że zajmuje około 15GB z 30GB.
Jakieś pomysły co mogę zrobić? Nie chce stracić danych z tej maszyny i stawiać wszystkich usług od nowa :/ 

```
"properties": {
        "title": "Reboot initiated from inside the machine",
        "details": "A reboot was triggered from inside the virtual machine. This could be due to a virtual machine operating system failure or as requested by an authorized user or process. The virtual machine will be back online after the reboot completes.",
        "currentHealthStatus": "Unavailable",
        "previousHealthStatus": "Unavailable",
        "type": "Downtime",
        "cause": "UserInitiated"
    },
```

**Comment by eklime** [comment]
Z poziomu portalu masz Run Command czy Serial Console - tamtędy próbuj.

**Comment by MarczakIO** [comment]
Tutaj ładnie opisują co można zrobić 

https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/troubleshoot-ssh-connection

**Comment by kamil-mrzyglod** [comment]
A czy maszynka w ogóle się bootuje?

**Comment by MariuszFerdyn** [comment]
W zależności jaka to dokładnie system operacyjny czy z Azure, czy twój własny upload.
Jeżeli z Azure tzw. Marketplace to powinien Ci działać Serial Console - o ile jądro się wczytało to już będziesz miał dostęp. https://docs.microsoft.com/pl-pl/azure/virtual-machines/troubleshooting/serial-console-windows

Jeżeli nie ma dostępu - to zawsze spróbuj Redeploy - ale zapewne nic to nie da. I pozostaje Ci:
1. Stworzyć maszyne wirtualną V3, gdzie jest nested Virtualisation
2. Zainstalować Hyper-V, zrobić sobie sięć na maszynie docelowej o ile potrzebna https://rzetelnekursy.pl/nat-on-windows-2016-or-on-windows-10-quick-config/
3. Ściągnać dysk VHD z Azure (bez Generalize) https://docs.microsoft.com/en-us/azure/virtual-machines/windows/download-vhd
4. Zrobic maszyne wirtualną z tym dyskiem i popatrzeć co się dzieje, naprawić
5. Wgrać VHD na Azure i stworzyć naprawioną wirtualkę.

Punkt 1,2 można pominąć i wykorzystać swojego laptopa, o ile ma się dobre łącze.

**Comment by milons** [comment, marked as answer]
Po powrocie z urlopu przypomniałem sobie, że nie dałem znać o co chodziło ;) 
Maszyna nie odpowiadała, bo dostała jakiś update (wcześniej lub wtedy), który popsuł GRUBa.
Azure restartował tę maszynę i już nie wstała. 
Dowiedziałem się co się stało, gdy włączyłem Boot Diagnostic Screenshot.

Rozwiązanie problemu opisane było tutaj:
https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/1889509
https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/1889509/comments/16
